core/app/models/spree/variant.rb

```diff
@@ -350,6 +350,16 @@ def variant_properties
       end.flatten.compact
     end
 
+    # An Enumerable of media that goes along with a variant
+    # Stores may not necessarily want to attach images to variants
+    # This provides a more robust interface
+    # NOTE: In the future this will return an empty array
```

Comment:
Why would this be useful? What is a developer supposed to do with this information.

I assume this is for extension points, since we want to avoid `class_eval`ing, maybe we should look at introducing a `Spree::Config.variant_gallery_class` and a `Spree::Config.product_gallery_class`.

Reply:
I like that idea of configurable image attachment classes. If we want extensions to be able to hook into we want to provide a real integration point. 

Also I don’t think `gallery` is the correct term. `gallery_images` would follow the “plural is a collection” convention of Rails better. We could even override the `images` getter. Btw. We need to deprecate the `images =` setter as well or delegate this to the images provider class.